sale_order_general_discount_type/models/sale_order.py

```
@@ -0,0 +1,105 @@
+
+    def _get_hidden_discount_cleanup_vals(self) -> dict[str, float]:
+        """Return values needed to clear the hidden discount field.
+
+        :return: Values for the non-selected discount field.
+        """
+        self.ensure_one()
+        if self.general_discount_type == "fixed":
+            return {"general_discount": 0.0}
+        return {"general_discount_fixed": 0.0}
+
+    def _cleanup_hidden_discount_field(self, use_write: bool = True):
```

Ruff (0.15.10)

[warning] Missing return type annotation for private function `_cleanup_hidden_discount_field`

Add return type annotation: `None`

(ANN202)